pkg/webhook/workloadspread/validating/workloadspread_validation.go

```diff
@@ -240,6 +271,35 @@ func validateWorkloadSpreadSubsets(ws *appsv1alpha1.WorkloadSpread, subsets []ap
 		}
 
 		//TODO validate patch
+		if subset.Patch.Raw != nil && workloadTemplate != nil {
+			var podSpec v1.PodTemplateSpec
+			switch workloadTemplate.GetObjectKind().GroupVersionKind() {
+			case controllerKruiseKindCS:
+				podSpec = workloadTemplate.(*appsv1alpha1.CloneSet).Spec.Template
+			case controllerKindDep:
+				podSpec = workloadTemplate.(*appsv1.Deployment).Spec.Template
+			case controllerKindRS:
+				podSpec = workloadTemplate.(*appsv1.ReplicaSet).Spec.Template
+			case controllerKindJob:
+				podSpec = workloadTemplate.(*batchv1.Job).Spec.Template
+			case controllerKindSts:
+				podSpec = workloadTemplate.(*appsv1.StatefulSet).Spec.Template
+			}
+			podBytes, _ := json.Marshal(podSpec)
+			modified, err := strategicpatch.StrategicMergePatch(podBytes, subset.Patch.Raw, &v1.Pod{})
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("patch"), string(subset.Patch.Raw), fmt.Sprintf("failed to merge patch: %v", err)))
+			}
+			newPod := &v1.Pod{}
+			if err = json.Unmarshal(modified, newPod); err != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("patch"), string(subset.Patch.Raw), fmt.Sprintf("failed to unmarshal: %v", err)))
+			}
+			coreNewPod, CovErr := convertor.ConvertPod(newPod)
+			if CovErr != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("patch"), newPod, fmt.Sprintf("Convert_v1_Pod_To_core_Pod failed: %v", err)))
+			}
+			allErrs = append(allErrs, corevalidation.ValidatePodSpec(&coreNewPod.Spec, &coreNewPod.ObjectMeta, fldPath.Index(i).Child("patch"), webhookutil.DefaultPodValidationOptions)...)
+		}
```

Comment:
how about the case that workloadTemplate is nil ? in such case, we cannot use ValidatePodSpec directly, instead, we should check whether the patch had change some fields that should not be changed such as affinity

Reply:
Yes. I'll add validation for this case.